flex/loading/schema/paths/path_item/operation/responses/single/schema/ref.py

```diff
@@ -24,6 +26,12 @@
 def validate_reference(reference, context, **kwargs):
     try:
         parts = urlparse.urlparse(reference)
+        if parts.path:
+            from flex.core import load_source
+            if parts.path.startswith('/'):
+                context = load_source(parts.path)
+            elif 'base_path' in kwargs:
+                context = load_source(os.path.join(kwargs['base_path'], parts.path))
```

Comment:
This same block of code is repeated quite a few times.  Can you abstract it into a reusable function.